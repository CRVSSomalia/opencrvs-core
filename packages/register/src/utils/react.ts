export type ComponentProps<C> = C extends React.Component<infer P, any>
  ? P
  : never
