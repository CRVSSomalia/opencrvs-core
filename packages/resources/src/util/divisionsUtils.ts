import { ILocation } from 'src/features/administrative/service'

export const appendDivisions = (
  divisions: ILocation[],
  newDivisions: ILocation[]
): ILocation[] => {
  return newDivisions.reduce((total, item) => {
    total.push(item)
    return total
  }, divisions)
}
