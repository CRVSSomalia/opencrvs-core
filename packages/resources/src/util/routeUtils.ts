export const replaceParamsInRoute = (
  route: string,
  params: string[]
): string => {
  return route.replace(/%(\d+)/g, (_, n) => params[+n - 1])
}
