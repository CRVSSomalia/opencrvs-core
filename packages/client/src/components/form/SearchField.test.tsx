/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
 * graphic logo are (registered/a) trademark(s) of Plan International.
 */
import * as React from 'react'
import { IntlProvider } from 'react-intl'
import { createShallowRenderedComponent } from '@client/tests/util'
import { SearchField } from './SearchField'

describe('Search Field component', () => {
  const mock = jest.fn()
  const testComponent = createShallowRenderedComponent(
    <IntlProvider locale="en">
      <SearchField
        fieldName="registrationOffice"
        fieldValue=""
        onModalComplete={mock}
        searchableResource="locations"
        searchableType="ADMIN_STRUCTURE"
      />
    </IntlProvider>
  )

  it('renders without crashing', () => {
    expect(testComponent).toMatchSnapshot()
  })
})
