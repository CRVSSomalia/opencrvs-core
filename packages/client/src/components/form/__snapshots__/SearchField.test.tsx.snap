// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Field component renders without crashing 1`] = `ShallowWrapper {}`;
